src/main/java/redis/clients/jedis/JedisClusterCommand.java

```diff
@@ -145,18 +145,29 @@ private T runWithRetries(final int slot) {
         releaseConnection(connection);
       }
       if (Instant.now().isAfter(deadline)) {
-        throw new JedisClusterMaxAttemptsException("Deadline exceeded"); // TODO: change to JedisClusterException
+        // TODO: change to JedisClusterOperationException or a new sub-class of it
+        throw new JedisClusterMaxAttemptsException("Deadline exceeded");
       }
     }
 
     throw new JedisClusterMaxAttemptsException("No more cluster attempts left.");
   }
 
+  /**
+   * Related values should be reset if <code>TRUE</code> is returned.
+   *
+   * @param attemptsLeft
+   * @param consecutiveConnectionFailures
+   * @param doneDeadline
+   * @return true - if some actions are taken
+   * <br /> false - if no actions are taken
+   */
   private boolean handleConnectionProblem(int attemptsLeft, int consecutiveConnectionFailures, Instant doneDeadline) {
     if (this.maxAttempts < 3) {
       // keep the old implementations as is?
       if (attemptsLeft == 0) {
         this.connectionHandler.renewSlotCache();
+        return true;
```

Comment:
At this point there are no more attempts left (`attemptsLeft == 0`), so it doesn't matter what we return here.

Reply:
I know. But I wanted to maintain `true - if some actions are taken`.